prov/rxm/src/rxm_ep.c

```diff
@@ -327,7 +327,13 @@ static void rxm_recv_queue_close(struct rxm_recv_queue *recv_queue)
 	if (recv_queue->fs) {
 		rxm_recv_fs_free(recv_queue->fs);
 	}
-	// TODO cleanup recv_list and unexp msg list
+
+	while (!dlist_empty(&recv_queue->recv_list)) {
+		dlist_remove(&recv_queue->recv_list);
+	}
+	while (!dlist_empty(&recv_queue->unexp_msg_list)) {
+		dlist_remove(&recv_queue->unexp_msg_list);
+	}
```

Comment:
Try moving this code to before the rxm_recv_fs_free call. The entries that are stored in the dlist are part of the freestack that is freed there